Tasks/MSBuild/Helpers.ps1

```diff
@@ -60,6 +60,11 @@ function Invoke-BuildTools {
         [switch]$NoTimelineLogger
     )
 
+    if ($NuGetRestore)
+    {
+        Write-Warning (Get-LocalizedString -Key "The `"Restore NuGet Packages`" option is deprecated. To restore NuGet packages in your build, add a NuGet Installer task to your build definition.")
```

Comment:
I'm curious, why this change? Is this to avoid updating code in multiple places? If so I've added a mechanism recently for common code to be shared across tasks.

If you make this change, then you need to check to make sure the template corresponds with the change here.